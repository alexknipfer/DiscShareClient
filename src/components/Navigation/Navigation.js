import { Link } from 'react-router-dom'
import { Menu } from 'semantic-ui-react'
import React from 'react'

const Navigation = () => {
  return (
    <div>
      <Menu pointing fixed="top">
        <Link to="/">
          <Menu.Item link>Home</Menu.Item>
        </Link>
        <Link to="/register">
          <Menu.Item link>Register</Menu.Item>
        </Link>
      </Menu>
    </div>
  )
}

export default Navigation
