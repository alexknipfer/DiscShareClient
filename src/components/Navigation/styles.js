import { Menu } from 'semantic-ui-react'
import styled from 'styled-components'

export const Nav = styled(Menu)`background-color: #253237 !important;`
