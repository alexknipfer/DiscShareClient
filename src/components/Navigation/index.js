import Navigation from './Navigation'
import { withRouter } from 'react-router-dom'

export default withRouter(Navigation)
