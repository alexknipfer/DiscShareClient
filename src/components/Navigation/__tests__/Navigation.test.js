import React from 'react'
import Navigation from '../Navigation'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'
import { shallow } from 'enzyme'
import { Link } from 'react-router-dom'
import { Menu } from 'semantic-ui-react'

describe('Testing Navigation', () => {
  it('Render nav with auth', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/login']}>
          <Navigation auth={true} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Render nav without auth', () => {
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={['/login']}>
          <Navigation auth={false} />
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Has three links', () => {
    const wrapper = shallow(<Navigation auth={true} />)
    expect(wrapper.find(Link).length).toBe(2)
  })

  it('Simulate Click', () => {
    const wrapper = shallow(<Navigation auth={true} />)
    wrapper
      .find(Menu)
      .last()
      .simulate('click')
  })
})
