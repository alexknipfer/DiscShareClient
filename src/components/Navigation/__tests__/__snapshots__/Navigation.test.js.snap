// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Navigation Render nav with auth 1`] = `
<div>
  <div
    className="ui inverted pointing top fixed menu"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="link item"
        onClick={[Function]}
      >
        Home
      </div>
    </a>
    <a
      href="/editAccount"
      onClick={[Function]}
    >
      <div
        className="link item"
        onClick={[Function]}
      >
        Edit Account
      </div>
    </a>
    <a
      className="link item"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>
</div>
`;

exports[`Testing Navigation Render nav without auth 1`] = `
<div
  className="ui inverted pointing top fixed sc-bdVaJa iCWLKP menu"
>
  <a
    href="/"
    onClick={[Function]}
  >
    <div
      className="link item"
      onClick={[Function]}
    >
      Home
    </div>
  </a>
  <a
    href="/register"
    onClick={[Function]}
  >
    <div
      className="link item"
      onClick={[Function]}
    >
      Register
    </div>
  </a>
  <a
    href="/login"
    onClick={[Function]}
  >
    <div
      className="link item"
      onClick={[Function]}
    >
      Login
    </div>
  </a>
</div>
`;
